base/transform_apply.go

```diff
@@ -0,0 +1,74 @@
+package base
+
+import (
+	"context"
+	"io"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/ioes"
+	"github.com/qri-io/qfs"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/qri-io/qri/repo"
+	reporef "github.com/qri-io/qri/repo/ref"
+	"github.com/qri-io/qri/startf"
+)
+
+// TODO(dustmop): Tests. Especially once the `apply` command exists.
+
+// TransformApply applies the transform script to order to modify the changing dataset
+func TransformApply(ctx context.Context, ds *dataset.Dataset, r repo.Repo, str ioes.IOStreams, scriptOut io.Writer, secrets map[string]string) error {
+	pro, err := r.Profile()
+	if err != nil {
+		return err
+	}
+
+	headPath := ""
+	if ds.Name != "" {
+		// Lookup the dataset to retrieve the head version
+		lookup := &reporef.DatasetRef{Peername: pro.Peername, Name: ds.Name}
+		err = repo.CanonicalizeDatasetRef(r, lookup)
+		if err == repo.ErrNotFound || err == repo.ErrNoHistory {
+			// Dataset either does not exist yet, or has no history. Not an error.
+		} else if err != nil {
+			return err
+		}
+		headPath = lookup.Path
+	}
+
+	target := ds
+	head := &dataset.Dataset{}
+
+	if headPath != "" {
+		// Load the dataset's most recent version
+		if head, err = dsfs.LoadDataset(ctx, r.Store(), headPath); err != nil {
+			return nil
+		}
+		if head.BodyPath != "" {
+			var body qfs.File
+			body, err = dsfs.LoadBody(ctx, r.Store(), head)
+			if err != nil {
+				return err
+			}
+			head.SetBodyFile(body)
+		}
+	}
```

Comment:
I'm assuming the plan is to replace this when we rebase #1361?

Reply:
Yes, I think this is definitely a good place to use the loadDataset interface. No need for apply to know concretely how a dataset is being loaded.